examples/step-27/step-27.cc

```diff
@@ -417,9 +350,11 @@ namespace Step27
       solution,
       estimated_error_per_cell);
 
-
-    Vector<float> smoothness_indicators(triangulation.n_active_cells());
-    estimate_smoothness(smoothness_indicators);
+    // Estimating the smoothness is performed with the method of decaing
+    // expansion coefficients as outlined in the introduction.
+    Vector<float> smoothness_indicators;
+    SmoothnessEstimator::estimate_by_coeff_decay<FESeries::Fourier<dim>>(
```

Comment:
One option is to take `FESeries::Fourier` so that all the setup related to various FEs inside it are done once. You would also have a much better control over quadrature rule used there as well as the maximum degree.